tools/build-config

```diff
@@ -6,31 +6,50 @@ const fs = require('fs')
 const flat = require('flat')
 const decamelize = require('decamelize')
 
-program
-  .option('-d, --debug', 'enable extreme logging')
-  .requiredOption('--directory <directory>', 'the directory to put the created xcconfig file in')
-
-const createXcconfigs = async (buildConfigName, directory) => {
+const loadBuildConfigAsXCConfig = (buildConfigName, directory) => {
   const buildConfig = loadBuildConfig(buildConfigName)
   const xcconfigOptions = flat(buildConfig, {
     delimiter: '_',
     transformKey: key => decamelize(key).toUpperCase()
   })
   const prefixed = Object.keys(xcconfigOptions).map(key => `BUILD_CONFIG_${key} = ${xcconfigOptions[key]}`)
   prefixed.push(`BUILD_CONFIG_NAME = ${buildConfigName}`)
-  fs.writeFileSync(`${directory}/buildConfig.tmp.xcconfig`, prefixed.join('\n'))
+  return prefixed.join('\n')
 }
 
 program
-  .command('config <build_config_name>')
+  .command('write-xcconfig <build_config_name>')
+  .requiredOption('--directory <directory>', 'the directory to put the created xcconfig file in')
   .description('create and write a new .xcconfig from the iosBuildOptions of the specified build config')
-  .action(async buildConfigName => {
+  .action((buildConfigName, cmdObj) => {
     try {
-      await createXcconfigs(buildConfigName, program.directory)
+      const xcconfig = loadBuildConfigAsXCConfig(buildConfigName)
+      fs.writeFileSync(`${cmdObj.directory}/buildConfig.tmp.xcconfig`, xcconfig)
     } catch (e) {
       console.error(e)
       process.exit(1)
     }
   })
 
+program
+  .command('to-xcconfig <build_config_name>')
```

Comment:
Is this code supposed to resolve the windows config generation?  I can't get it to work on windows...

Reply:
No, this is only a minor refactor.

To get the build working on windows we need to write a batch script to call JS code. https://issues.tuerantuer.org/browse/NATIVE-608

Unfortunately I can not work on this as I do not have a windows installation at hand.